zulipterminal/model.py

```diff
@@ -560,10 +561,16 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
-    def get_file_upload_uri(self, file_location: str) -> str:
-        with open(file_location, "rb") as fp:
-            result = self.client.upload_file(fp)
-        return result["uri"]
+    def get_file_upload_uri(self, file_location: str) -> Optional[str]:
+        if os.path.exists(file_location):
+            with open(file_location, "rb") as fp:
```

Comment:
Rereading this, it would definitely be better to use the exception approach here, and focus error handling for upload as occurring from the model, compared to error handling with accessing the file from the calling function.

It's possible there could be errors related to processing the upload related to the file handling, but that would be part of the upload process.